test/e2e/qatplugin_test.go

```diff
@@ -0,0 +1,71 @@
+// Copyright 2020 Intel Corporation. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e_test
+
+import (
+	"time"
+
+	"github.com/onsi/ginkgo"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2epod "k8s.io/kubernetes/test/e2e/framework/pod"
+	imageutils "k8s.io/kubernetes/test/utils/image"
+)
+
+const (
+	pluginYaml = "deployments/qat_plugin/qat_plugin_kernel_mode.yaml"
+)
+
+func init() {
+	ginkgo.Describe("QAT plugin in kernel mode", describeQatplugin)
+}
+
+func describeQatplugin() {
+	f := framework.NewDefaultFramework("qatplugin")
+
+	yamlPath, err := locateRepoFile(pluginYaml)
+	if err != nil {
+		framework.Failf("unable to locate %q: %v", pluginYaml, err)
+	}
+
+	ginkgo.It("does something", func() {
+		ginkgo.By("deploying QAT plugin in kernel mode")
+		framework.ExpectNoError(runCommand(framework.TestContext.KubectlPath, "--namespace", f.Namespace.Name, "create", "-f", yamlPath))
+
+		ginkgo.By("waiting for QAT plugin's availability")
+		if _, err := e2epod.WaitForPodsWithLabelRunningReady(f.ClientSet, f.Namespace.Name,
+			labels.Set{"app": "intel-qat-kernel-plugin"}.AsSelector(), 1 /* one replica */, 10*time.Second); err != nil {
+			framework.DumpAllNamespaceInfo(f.ClientSet, f.Namespace.Name)
+			framework.LogFailedContainers(f.ClientSet, f.Namespace.Name, framework.Logf)
+			framework.Failf("unable to wait for all pods to be running and ready: %v", err)
+		}
+
+		ginkgo.By("submitting a pod requesting QAT resources")
```

Comment:
`ginkgo.By("checking nodes' QAT resources")` before deploying an app? Could be a common func that can be reused by all the plugins.

Reply:
Successful completion of the pods requesting the resources is supposed to do exactly that. But checking nodes resources explicitly won't harm. Will take a look.

Reply:
my initial thinking was it's better to check that first and only deploy the pod and pull the container if the node is OK.  This would also avoid the pod pending timeout to trigger

Reply:
Added `WaitForNodesWithResource()` for this purpose.